### PR TITLE
add plugin support

### PR DESCRIPTION
**Pull request by bilsch**
hiera config to add / remove plugins also works

kibana::plugins:
  marvel:
    source: elasticsearch/marvel/latest

And to remove

kibana::plugins:
  marvel:
    source: elasticsearch/marvel/latest
    ensure: absent

### CODE REVIEW

File: manifests/plugin.pp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      $name_file_path = "${plugins_dir}/${base_module_name}/.name"
+      exec {"install_plugin_${name}":
+        command => $install_cmd,
+        creates => $name_file_path,
+        notify  => Service['kibana'],
+        require => File[$plugins_dir],
+      }
+      file {$name_file_path:
+        ensure  => file,
+        content => "${base_module_name}", # lint:ignore:only_variable_string
```
Comment (dansajner) [inline manifests/plugin.pp]: Why does this need to be quoted?

**Comment by dansajner** [comment]
I'm concerned that this will obscure the upstream pull request which was originally scoped to just fix systemd issues.

File: manifests/plugin.pp (hunk 1 above)
Comment (bilsch) [inline manifests/plugin.pp]: was not paying attention, yea no reason to quote that one

**Comment by bilsch** [comment]
the only reason I made this change as a branch off of shield is that I did not want to lose what Jesse had. If his change merges upstream then I can rebase everything. The intent long-term is to get this change merged in - the plugin support is not at all there in upstream yet so...

**Comment by dansajner** [comment]
I guess I'd like to see our master branch updated with Jesse's changes and then these changes and we'll just operate off of the master branch of our fork.

**Comment by adamcstephens** [comment]
To me its cleaner if master tracks upstream/master, and we have a separate branch that we're using.

**Comment by bilsch** [comment]
there is an outstanding pull request from Jesse to the evenup guys. Once that is merged technically that would be the case. ( Jesse's change was made against my shield branch and merged so... )

I can merge his change to master in our fork and then rebase this on top of our master but that will result in more fun rebase work later potentially ( for example if Jesse re-works his patch to get it accepted then I have to rebase our master to match )

**Comment by bilsch** [comment]
to be clear, my proposal is to get Jesse's change merged to upstream/master, then sync up our master and rebase this branch

**Comment by bilsch** [comment]
closing this pull request, will rebase against master and send PR to upstream

**Pull request closed without merging**